extensions/clusters/clusters.go

```diff
@@ -824,32 +825,19 @@ func AgentAffinityConfigHelper(advancedClusterAffinity *management.Affinity) *co
 	return agentAffinity
 }
 
-// HardenK3SClusterConfig is a constructor for a apisV1.Cluster object, to be used by the rancher.Client.Provisioning client.
+// HardenK3SClusterConfig is a function that modifies the cluster configuration to be hardened according to the CIS benchmark.
 func HardenK3SClusterConfig(clusterName, namespace string, clustersConfig *ClusterConfig, machinePools []apisV1.RKEMachinePool, cloudCredentialSecretName string) *apisV1.Cluster {
 	v1Cluster := NewK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, machinePools, cloudCredentialSecretName)
+	v1Cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName = rancherRestricted
 
-	if clustersConfig.KubernetesVersion <= string(provisioninginput.PSPKubeVersionLimit) {
-		v1Cluster.Spec.RKEConfig.MachineGlobalConfig.Data["kube-apiserver-arg"] = []string{
-			"enable-admission-plugins=NodeRestriction,PodSecurityPolicy,ServiceAccount",
-			"audit-policy-file=/var/lib/rancher/k3s/server/audit.yaml",
-			"audit-log-path=/var/lib/rancher/k3s/server/logs/audit.log",
-			"audit-log-maxage=30",
-			"audit-log-maxbackup=10",
-			"audit-log-maxsize=100",
-			"request-timeout=300s",
-			"service-account-lookup=true",
-		}
-	} else {
-		v1Cluster.Spec.RKEConfig.MachineGlobalConfig.Data["kube-apiserver-arg"] = []string{
-			"admission-control-config-file=/var/lib/rancher/k3s/server/psa.yaml",
```

Comment:
was this not supposed to be here before, or did we miss moving it over? (why was this rm'd, since it used to be in the 'else' block)

Reply:
This is being removed because Rancher sets this when you set the PSACT. It's redundant having it here.